ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -41,6 +41,41 @@ EOF
     fi
 }
 
+function update_cloud_credentials(){
+    platform=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')
+    cco_mode=$(oc get cloudcredential cluster -o jsonpath='{.spec.credentialsMode}')
+    if (( SOURCE_MINOR_VERSION == TARGET_MINOR_VERSION )) || (( SOURCE_MINOR_VERSION < 8 )) || [[ "${cco_mode}" != "Manual" ]]; then
+        echo "No need update cloud credentials" && return
+    fi
+    preCredsDir="/tmp/pre-include-creds"
+    tobeCredsDir="/tmp/tobe-include-creds"
+    mkdir "${preCredsDir}" "${tobeCredsDir}"
+    # Extract all CRs from live cluster with --included
+    if ! oc adm release extract --to "${preCredsDir}" --included --credentials-requests; then
+        echo "Failed to extract CRs from live cluster!" && exit 1
+    fi
+    if ! oc adm release extract --to "${tobeCredsDir}" --included --credentials-requests "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"; then
+        echo "Failed to extract CRs from tobe upgrade release payload!" && exit 1
+    fi
+
+    # TODO: add gcp and azure
+    # Update iam role with ccoctl based on tobeCredsDir
+    if [[ "${platform}" == "AWS" ]]; then
+        if [[ ! -e ${SHARED_DIR}/aws_oidc_provider_arn ]]; then
+            echo "No aws_oidc_provider_arn file in SHARED_DIR" && exit 1
+        else
+            oidc_provider=$(head -n1 ${SHARED_DIR}/aws_oidc_provider_arn)
+        fi
+    fi
+    if ! diff -r "${preCredsDir}" "${tobeCredsDir}" &> /dev/null; then
```

Comment:
Is this a restrict diff? Even just some permission difference, would start to create new credentials?

Reply:
yes, it will

Reply:
so that would generate some duplicates IAM credentials, when only some permission difference between source and target payload?

Reply:
If my understanding correct, according to the discussion with jianping in slack, ccoctl will judge current role and only update necessary ones.

Reply:
Yeah, If I recall correctly, oc outputs all the CR files needed by the target image instead of the different files as compared with source image. And ccoctl is able to apply the differences to the cluster when we specify the target CR set